persistence/provider/boltdb/eventstore.go

```diff
@@ -110,6 +127,64 @@ func (r *eventStoreRepository) NextEventOffset(
 	return next, err
 }
 
+// LoadEventsBySource loads the events produced by a specific handler.
+//
+// hk is the handler's identity key.
+//
+// id is the instance ID, which must be empty if the handler type does not
+// use instances.
+//
+// m is ID of a "barrier" message. If supplied, the results are limited to
+// events with higher offsets than the barrier message. If the message
+// cannot be found, UnknownMessageError is returned.
+func (r *eventStoreRepository) LoadEventsBySource(
+	ctx context.Context,
+	hk, id, d string,
+) (eventstore.Result, error) {
+	var (
+		o   uint64
+		err error
+	)
+
+	if d != "" {
+		r.db.View(
+			ctx,
+			func(tx *bbolt.Tx) {
+				if messageIDs, exists := bboltx.TryBucket(
+					tx,
+					r.appKey,
+					eventStoreBucketKey,
+					eventStoreMessageIDsBucketKey,
+				); exists {
+					v := messageIDs.Get([]byte(d))
+					if v == nil {
+						err = &eventstore.UnknownMessageError{
+							MessageID: d,
+						}
+
+						return
+					}
+
+					o = unmarshalUint64(v) + 1
```

Comment:
🐧 

```suggestion
					// Increment the offset to match all messages
					// after the barrier message exclusively.
					o = unmarshalUint64(v) + 1
```

Just to make this match the SQL impl - you had the comment explaining why there, and I thought that was good.